src/python/espressomd/reaction_ensemble.pyx

```diff
@@ -368,8 +373,8 @@ cdef class ConstantpHEnsemble(ReactionAlgorithm):
             Sets the pH that the method assumes for the implicit pH bath.
 
             """
-            if(pH<=0):
-                raise ValueError("pH must be strictly positive")
+            if(pH<4 or pH >10):
+                print("Be extremely careful in this extreme pH range when using the constant pH method. The ions which make up the pH (H+ or OH-) are not included in the method. Therefore the method might get screening effects wrong. See paper: Simulation of weak polyelectrolytes: a comparison between the constant pH and the reaction ensemble method by Landsgesell et al.")
```

Comment:
This is pretty vague..

Reply:
True. When exactly the problems with constant pH start will depend on the exact system that you are looking at and also at the salt concentration that you set there.

Reply:
I think this belongs into the documentation and not into a warning. If the method does not work as advertised with those parameters, it should not run, otherwise it should do as it is told. We should consider to make it a policy not to do warnings. Either something is impossible or likely unintended, then it should throw and make the user opt-in explicitly.

Reply:
Im also against warnings!

Reply:
Well the algorithm works for any input. It's just you will very likely produce very wrong physics. I am in favour of this warning. People made this wrong for over 20 years. I see a need to warn them here.

Reply:
In addition you now allow negative pH values (it does not raise an error anymore)